src/addons/meta/cursor.c

```diff
@@ -624,6 +634,13 @@ int ecs_meta_pop(
             ecs_assert(op->kind == EcsOpPop, ECS_INTERNAL_ERROR, NULL);
         } else if (op->kind == EcsOpArray || op->kind == EcsOpVector) {
             /* Collection type, nothing else to do */
+            
+            /* If no values were serialized for scope, fini vector memory */
```

Comment:
This is debatable, if the deserialize instructions contained a push + pop (e.g. `[]`) the result should be an initialized but empty vector.